### PR TITLE
[FLINK-26373][connector/kinesis] Renaming KinesisDataStreams module to KinesisStreams

### PR DESCRIPTION
**Pull request by vahmed-hamdy**
## What is the purpose of the change

- Renaming `KinesisDataStreams` to `KinesisStreams` and `flink-connector-aws-kinesis-data-streams` to  `flink-connector-aws-kinesis-streams` following the firehose module.

## Brief change log

- Renamed `flink-connector-aws-kinesis-data-streams` to  `flink-connector-aws-kinesis-streams`.
- Renamed classes matching`KinesisDataStreams*` to `KinesisStreams*`
- Renamed dependant modules `flink-sql-connector-aws-kinesis-data-streams` to  `flink-sql-connector-aws-kinesis-streams` and `flink-end-to-end-tests-kinesis-data-streams` to  `flink-end-to-end-tests-kinesis-streams` 

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency):no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`:yes
  - The serializers: no
  - The runtime per-record code paths (performance sensitive):no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper:no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? docs

### CODE REVIEW

**Comment by vahmed-hamdy** [comment]
@flinkbot run azure

**Comment by vahmed-hamdy** [comment]
@flinkbot run azure

**Comment by vahmed-hamdy** [comment]
@flinkbot run azure

**Comment by vahmed-hamdy** [comment]
@flinkbot run azure

File: flink-connectors/flink-connector-aws-kinesis-firehose/src/main/java/org/apache/flink/connector/firehose/table/KinesisFirehoseDynamicTableFactory.java [hunk 1]
```diff
@@ -45,8 +45,8 @@ public DynamicTableSink createDynamicTableSink(Context context) {
 
         AsyncDynamicSinkContext factoryContext = new AsyncDynamicSinkContext(this, context);
 
-        KinesisFirehoseDynamicSink.KinesisDataFirehoseDynamicSinkBuilder builder =
-                new KinesisFirehoseDynamicSink.KinesisDataFirehoseDynamicSinkBuilder();
+        KinesisFirehoseDynamicSink.KinesisFirehoseDynamicSinkBuilder builder =
+                new KinesisFirehoseDynamicSink.KinesisFirehoseDynamicSinkBuilder();
```
Comment (CrynetLogistics) [inline flink-connectors/flink-connector-aws-kinesis-firehose/src/main/java/org/apache/flink/connector/firehose/table/KinesisFirehoseDynamicTableFactory.java:49]: Nice catch

**Review by CrynetLogistics — APPROVED** [review]

**Comment by vahmed-hamdy** [comment]
@flinkbot run azure

File: flink-connectors/flink-connector-aws-kinesis-streams/pom.xml [hunk 2]
```diff
@@ -30,7 +30,7 @@ under the License.
 		<relativePath>..</relativePath>
 	</parent>
 
-	<artifactId>flink-connector-aws-kinesis-data-streams</artifactId>
+	<artifactId>flink-connector-aws-kinesis-streams</artifactId>
 	<name>Flink : Connectors : AWS Kinesis Data Streams</name>
```
Comment (dannycranmer) [inline flink-connectors/flink-connector-aws-kinesis-streams/pom.xml:34]: Missed "Data" here

**Review by dannycranmer — CHANGES_REQUESTED** [review]

**Pull request merged**